### PR TITLE
Add token support to Tentacle

### PR DESCRIPTION
**Pull request by scme0**
# Background

As part of work on the Kubernetes Agent work, we are updating Tentacle to take a JWT Token for authentication instead of Username and Password or API Key.

[sc-60667]

# Results

I've added a `--bearerToken` command line option so that we can specify it instead of an API Key.

I've manually tested this by installing a Tentacle Target in kubernetes for both linux and windows so the `configure-tentacle.sh` and `.ps1` scripts have been tested 👍 

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle/Octopus.Tentacle.csproj [hunk 1]
```diff
@@ -41,7 +41,7 @@
 		<DefineConstants>$(DefineConstants);HTTP_CLIENT_SUPPORTS_SSL_OPTIONS;REQUIRES_EXPLICIT_LOG_CONFIG;REQUIRES_CODE_PAGE_PROVIDER;USER_INTERACTIVE_DOES_NOT_WORK;DEFAULT_PROXY_IS_NOT_AVAILABLE;HAS_NULLABLE_REF_TYPES</DefineConstants>
 	</PropertyGroup>
 	<ItemGroup>
-		<PackageReference Include="Octopus.Client" Version="14.3.884" />
+		<PackageReference Include="Octopus.Client" Version="14.3.892-scme-add-support-for-token-authentication" />
```
Comment (APErebus) [inline source/Octopus.Tentacle/Octopus.Tentacle.csproj:44]: This should be a proper release right?

File: source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs [hunk 2]
```diff
@@ -218,7 +218,7 @@ Uri GetActiveTentacleAddress()
                     serverCommsPort ??= serverCommsAddressUri.Port;
                 }
 
-                return new Uri($"https://{serverCommsAddressUri.Host}:{serverCommsPort}");
+                return new Uri($"https://{serverCommsAddressUri?.Host}:{serverCommsPort}");
```
Comment (APErebus) [inline source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs:221]: I'm assuming this is possibly null as `api.ServerUri` might be null? If so, would it not be better to throw earlier and make sure it's not null at this point? having a null host here doesn't make sense

**Review by APErebus — CHANGES_REQUESTED** [review]
2 minor things, but the rest looks good!

File: source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs (hunk 2 above)
Comment (scme0) [reply source/Octopus.Tentacle/Commands/RegisterMachineCommandBase.cs]: Good point 👍 Fixed.

File: source/Octopus.Tentacle/Octopus.Tentacle.csproj (hunk 1 above)
Comment (scme0) [reply source/Octopus.Tentacle/Octopus.Tentacle.csproj]: Yep, fixed now 👍

**Review by APErebus — APPROVED** [review]
Excellent! Nice work!

**Pull request merged**